tests/components/broadlink/test_climate.py

```diff
@@ -0,0 +1,193 @@
+"""Tests for Broadlink climate."""
+
+from homeassistant.components.broadlink.climate import SensorMode
+from homeassistant.components.broadlink.const import DOMAIN
+from homeassistant.components.climate import (
+    ATTR_TEMPERATURE,
+    DOMAIN as CLIMATE_DOMAIN,
+    SERVICE_SET_TEMPERATURE,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.const import ATTR_ENTITY_ID, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.entity_component import async_update_entity
+
+from . import get_device
+
+
+async def test_climate_inner_heat_heating(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test Broadlink climate.
+
+    The test initialized with:
+    - SensorMode.INNER_SENSOR_CONTROL
+    - HVACMode.HEAT
+    - HVACAction.HEATING
+    """
+    device = get_device("Guest room")
+    mock_setup = await device.setup_entry(hass)
+
+    device_entry = device_registry.async_get_device(
+        identifiers={(DOMAIN, mock_setup.entry.unique_id)}
+    )
+    entries = er.async_entries_for_device(entity_registry, device_entry.id)
+    climates = [entry for entry in entries if entry.domain == Platform.CLIMATE]
+    assert len(climates) == 1
+
+    climate = climates[0]
+
+    mock_setup.api.get_full_status.return_value = {
+        "sensor": SensorMode.INNER_SENSOR_CONTROL,
```

Comment:
This is incorrect, as the data will not return an enum member defined in HA.
You probably wanted to return the value of the enum member like

```suggestion
        "sensor": SensorMode.INNER_SENSOR_CONTROL.value,
```